raiden/transfer/views.py

```diff
@@ -0,0 +1,273 @@
+# -*- coding: utf-8 -*-
+from raiden.transfer import channel
+from raiden.transfer.state import (
+    CHANNEL_STATE_OPENED,
+    CHANNEL_STATE_SETTLED,
+    NODE_NETWORK_UNKNOWN,
+    PaymentMappingState,
+)
+
+# TODO: Either enforce immutability or make a copy of the values returned by
+# the view functions
+
+
+def all_neighbour_nodes(node_state):
+    """ Return the identifiers for all nodes accross all payment networks which
+    have a channel open with this one.
+    """
+    addresses = set()
+
+    for payment_network in node_state.identifiers_to_paymentnetworks.values():
+        for token_network in payment_network.tokenidentifiers_to_tokennetworks.values():
+            for channel_state in token_network.partneraddresses_to_channels.values():
+                addresses.add(channel_state.partner_state.address)
+
+    return addresses
+
+
+def block_number(node_state):
+    return node_state.block_number
+
+
+def state_from_raiden(raiden):
+    return raiden.wal.state_manager.current_state
+
+
+def state_from_app(app):
+    return app.raiden.wal.state_manager.current_state
+
+
+def get_networkstatuses(node_state):
+    return node_state.nodeaddresses_to_networkstates
+
+
+def get_node_network_status(node_state, node_address):
+    return node_state.nodeaddresses_to_networkstates.get(
+        node_address,
+        NODE_NETWORK_UNKNOWN,
+    )
+
+
+def get_token_network_addresses_for(node_state, payment_network_id):
```

Comment:
What is the `payment_network_id`, the address of the registry contract?

Reply:
yes